src/chapters/databases-part2/game-logic.rst

```diff
@@ -0,0 +1,161 @@
+the following videos.
+
+Change Cells After Click
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube-nocookie.com/embed/an8sh-Xp5PQ" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
+Add Feedback Messages
+^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube-nocookie.com/embed/aIJZhKCjX1E" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
+``mines.html`` Summary
+----------------------
+
+#. Each time a player clicks a button on the game board, its appearance
+   changes.
+#. We use 5 session key/value pairs to determine the styling and content for
+   each cell:
+
+   a. ``flags``: This key points to a list of all the cells the player marked
+      with the *Flag Mine* option.
+   b. ``guesses``: Points to a list of safe cells the player has clicked.
+   c. ``mine_counts``: Points to a dictionary that contains the number of mines
+      surrounding each space on the board.
+   d. ``hit_mine``: A boolean value. Set to ``True`` when the player clicks on
+      a mine.
+   e. ``mines``: Points to a list containing the locations of hidden mines.
+
+#. By adding an extended ``if/elif/else`` block to ``mines.html``, we can check
+   the conditions for each cell just before it is rendered on the page.
+#. Adding another Jinja3 conditional inside the ``<header></header>`` element
+   allows us to display different feedback messages to the player.
+
+Test the Game!
+--------------
+
+Yay! Our Minesweeper game is ready. Play a few games to see how the application
+behaves.
+
+.. admonition:: Tip
+   
+   When your teacher and/or parents ask why you're playing games instead of
+   doing homework, tell them you're *beta-testing* a coding project.
+
+As you play, you will find some glitches and rough spots in the game. This is
+expected, and it's the whole point of testing! Keep a record of the flaws you
+find, and note how serious each one is. Errors that crash the application or
+impact the gameplay should be addressed before minor annoyances.
+
+Even though the game functions, we can't call the application "complete" just
+yet. In this chapter, we focused on getting the game up and running, and we
+accomplished that goal. However, the project does contain a number of
+shortcomings. How to fix these is up to you! You have a good starting place
+right now. Your ideas will determine the next step.
+
+.. admonition:: Note
+   
+   No project is ever really done. Unusual bugs will pop up that need to be
+   fixed, or new feature ideas will pop into your brain. Coding an application
+   is a continuous process of tinkering, testing, and discovery!
```

Comment:
Not sure if you would want to mention minimum viable product (MVP), it is something I we stressed a lot when teaching game design and apps. Maybe rework lines 151 to 155 just to introduce the term.  Students who are interested will find their way to google.

This was one of the videos we used https://www.youtube.com/watch?v=UvCri1tqIxQ.  Might be too much but I always thought exposing the students to industry thinking was good.  We talked a lot about scope and you've done a great job of showing how to create a game step by step!

Reply:
MVP was included in an earlier draft of the chapter (in the introduction, as it turns out), but I ultimately cut the definition and references.  As you said, it might be a bit too much.  There are so many details flying around in this chapter, that I wanted to avoid saturation.

However, your comment did allow me to see that **minimum viable product** was still in the index and led to a blank spot on the first page, so I removed it!  :-)